src/front/wgsl/mod.rs

```diff
@@ -498,16 +504,17 @@ impl BindingParser {
     }
 
     fn finish<'a>(self) -> Result<Option<crate::Binding>, Error<'a>> {
-        match (self.location, self.built_in, self.interpolation) {
-            (None, None, None) => Ok(None),
-            (Some(loc), None, interpolation) => {
-                Ok(Some(crate::Binding::Location(loc, interpolation)))
+        match (self.location, self.built_in, self.interpolation, self.sampling) {
+            (None, None, None, None) => Ok(None),
```

Comment:
We need to have a pass, after the module is mostly read, that would fill out the defaults where they are expected.
And we'll need to put a comment here saying that this is handled in this pass.